narwhals/dataframe.py

```diff
@@ -2022,6 +2073,61 @@ def head(self, n: int) -> Self:
         """
         return super().head(n)
 
+    def tail(self, n: int) -> Self:
+        r"""
+        Get the last `n` rows.
+
+        Arguments:
+            n: Number of rows to return.
+
+        Examples:
+            >>> import polars as pl
+            >>> import narwhals as nw
+            >>> lf_pl = pl.LazyFrame(
+            ...     {
+            ...         "a": [1, 2, 3, 4, 5, 6],
+            ...         "b": [7, 8, 9, 10, 11, 12],
+            ...     }
+            ... )
+            >>> lf = nw.LazyFrame(lf_pl)
+            >>> lframe = lf.tail(5).collect()
+            >>> lframe
```

Comment:
if we're adding a new docstring, can we follow the
- create pandas/polars dfs
- define a df-agnostic function
- show that both libraries produce the same output
pattern?

some old docstrings don't, but we're in the process of updating them